nBTC/sources/nbtc.move

```diff
@@ -172,20 +174,28 @@ public fun mint(
     let tx = tx::deserialize(&mut r);
 
     let tx_id = tx.tx_id();
+
+    // Double check prevent
+    assert!(!contract.tx_ids.contains(tx_id), ETxAlreadyUsed);
+    contract.tx_ids.add(tx_id, true);
+    // NOTE: We assume only one active key. We should handle mutiple nbtc active key in the
+    // future.
     let (mut amount, mut op_return) = verify_payment(
         light_client,
         height,
         proof,
         tx_index,
         &tx,
-        contract.nbtc_bitcoin_pkh,
+        contract.bitcoin_pkh,
```

Comment:
This is not what I meant. I will do a follow up PR - it will be faster.
Let's do together after you will merge this